### ISSUE TITLE
[🐛 Bug]: Kotlin/Selenium-Java 4.1.0 java.io.IOException: Cannot run program "hostname": error=2, No such file or directory

### ISSUE THREAD

**Issue by lroman-uz** [SeleniumHQ/selenium#10085]
### What happened?

We use Kotlin for our automation framework and since Selenium 4 was released we get an error (I attached the error log) when we try to execute our test in a cloud farm, Saucelabs, and we need to download some files using a precondition in a Windows machine. 
The error only happens using Kotlin, if we try to execute the same test in Java is working as expected so here is where I suspect there's an issue with the library and I need your help.
This error wasn't happening with Selenium 3.

### How can we reproduce the issue?

```shell
Here you have a repo with examples: https://github.com/lroman-uz/selenium4-saucelabserrors

Test failing is: Selenium4PreRunWindowsTest.kt

Same Test working in Java is: Selenium4JavaPreRunWindowsTest.java
```

### Relevant log output

```shell
nov. 25, 2021 10:15:37 A. M. org.openqa.selenium.net.HostIdentifier resolveHostName
ADVERTENCIA: Failed to resolve host name
java.io.IOException: Cannot run program "hostname": error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at java.base/java.lang.Runtime.exec(Runtime.java:590)
	at java.base/java.lang.Runtime.exec(Runtime.java:414)
	at java.base/java.lang.Runtime.exec(Runtime.java:311)
	at org.openqa.selenium.net.HostIdentifier.resolveHostName(HostIdentifier.java:50)
	at org.openqa.selenium.net.HostIdentifier.getHostName(HostIdentifier.java:116)
	at org.openqa.selenium.WebDriverException.getSystemInformation(WebDriverException.java:95)
	at org.openqa.selenium.WebDriverException.createMessage(WebDriverException.java:87)
	at org.openqa.selenium.WebDriverException.getMessage(WebDriverException.java:65)
	at java.base/java.lang.Throwable.getLocalizedMessage(Throwable.java:207)
	at java.base/java.lang.Throwable.toString(Throwable.java:355)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:544)
	at java.base/java.lang.Throwable.printStackTraceHelper(Throwable.java:342)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:318)
	at org.junit.runner.notification.Failure.getTrace(Failure.java:75)
	at com.intellij.junit4.JUnit4TestListener.getTrace(JUnit4TestListener.java:298)
	at com.intellij.junit4.JUnit4TestListener.testFailure(JUnit4TestListener.java:280)
	at com.intellij.junit4.JUnit4TestListener.testFailure(JUnit4TestListener.java:231)
	at com.intellij.junit4.JUnit4TestListener.testFailure(JUnit4TestListener.java:210)
	at org.junit.runner.notification.SynchronizedRunListener.testFailure(SynchronizedRunListener.java:63)
	at org.junit.runner.notification.RunNotifier$4.notifyListener(RunNotifier.java:142)
	at org.junit.runner.notification.RunNotifier$SafeNotifier.run(RunNotifier.java:72)
	at org.junit.runner.notification.RunNotifier.fireTestFailures(RunNotifier.java:138)
	at org.junit.runner.notification.RunNotifier.fireTestFailure(RunNotifier.java:132)
	at org.junit.internal.runners.model.EachTestNotifier.addFailure(EachTestNotifier.java:23)
	at org.junit.internal.runners.model.EachTestNotifier.addMultipleFailureException(EachTestNotifier.java:29)
	at org.junit.internal.runners.model.EachTestNotifier.addFailure(EachTestNotifier.java:21)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:329)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:319)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:250)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 41 more

nov. 25, 2021 10:15:38 A. M. org.openqa.selenium.net.HostIdentifier resolveHostAddress
ADVERTENCIA: Failed to resolve host address
java.lang.NullPointerException
	at org.openqa.selenium.net.HostIdentifier.resolveHostAddress(HostIdentifier.java:92)
	at org.openqa.selenium.net.HostIdentifier.getHostAddress(HostIdentifier.java:123)
	at org.openqa.selenium.WebDriverException.getSystemInformation(WebDriverException.java:95)
	at org.openqa.selenium.WebDriverException.createMessage(WebDriverException.java:87)
	at org.openqa.selenium.WebDriverException.getMessage(WebDriverException.java:65)
	at java.base/java.lang.Throwable.getLocalizedMessage(Throwable.java:207)
	at java.base/java.lang.Throwable.toString(Throwable.java:355)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:544)
	at java.base/java.lang.Throwable.printStackTraceHelper(Throwable.java:342)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:318)
	at org.junit.runner.notification.Failure.getTrace(Failure.java:75)
	at com.intellij.junit4.JUnit4TestListener.getTrace(JUnit4TestListener.java:298)
	at com.intellij.junit4.JUnit4TestListener.testFailure(JUnit4TestListener.java:280)
	at com.intellij.junit4.JUnit4TestListener.testFailure(JUnit4TestListener.java:231)
	at com.intellij.junit4.JUnit4TestListener.testFailure(JUnit4TestListener.java:210)
	at org.junit.runner.notification.SynchronizedRunListener.testFailure(SynchronizedRunListener.java:63)
	at org.junit.runner.notification.RunNotifier$4.notifyListener(RunNotifier.java:142)
	at org.junit.runner.notification.RunNotifier$SafeNotifier.run(RunNotifier.java:72)
	at org.junit.runner.notification.RunNotifier.fireTestFailures(RunNotifier.java:138)
	at org.junit.runner.notification.RunNotifier.fireTestFailure(RunNotifier.java:132)
	at org.junit.internal.runners.model.EachTestNotifier.addFailure(EachTestNotifier.java:23)
	at org.junit.internal.runners.model.EachTestNotifier.addMultipleFailureException(EachTestNotifier.java:29)
	at org.junit.internal.runners.model.EachTestNotifier.addFailure(EachTestNotifier.java:21)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:329)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Unable to parse remote response: 

	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:112)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:84)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:62)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:156)
	at org.openqa.selenium.remote.TracedCommandExecutor.execute(TracedCommandExecutor.java:50)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:559)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:246)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:168)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:146)
	at org.example.Selenium4PreRunWindowsTest.setUp(Selenium4PreRunWindowsTest.kt:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: org.openqa.selenium.json.JsonException: Unable to parse: 
	at org.openqa.selenium.json.Json.toType(Json.java:57)
	at org.openqa.selenium.json.Json.toType(Json.java:50)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:109)
	... 32 more
Caused by: org.openqa.selenium.json.JsonException: Expected to read a START_MAP but instead have: END. Last 0 characters read: 
Build info: version: '4.1.0', revision: '87802e897b'
System info: host: 'iMac-de-Luis.local', ip: '192.168.0.18', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.16', java.version: '14.0.2'
Driver info: driver.version: RemoteWebDriver
	at org.openqa.selenium.json.JsonInput.expect(JsonInput.java:307)
	at org.openqa.selenium.json.JsonInput.beginObject(JsonInput.java:229)
	at org.openqa.selenium.json.MapCoercer.lambda$apply$1(MapCoercer.java:64)
	at org.openqa.selenium.json.MapCoercer$$Lambda$93/0000000000000000.apply(Unknown Source)
	at org.openqa.selenium.json.JsonTypeCoercer.lambda$buildCoercer$6(JsonTypeCoercer.java:145)
	at org.openqa.selenium.json.JsonTypeCoercer$$Lambda$294/0000000000000000.apply(Unknown Source)
	at org.openqa.selenium.json.JsonTypeCoercer.coerce(JsonTypeCoercer.java:127)
	at org.openqa.selenium.json.Json.toType(Json.java:71)
	at org.openqa.selenium.json.Json.toType(Json.java:55)
	... 34 more
```

### Operating System

Windows 10/Saucelabs

### Selenium version

4.1.0

### What are the browser(s) and version(s) where you see this issue?

Chrome 93/Firefox 94/Edge 93

### What are the browser driver(s) and version(s) where you see this issue?

Ones Saucelabs is using

### Are you using Selenium Grid?

Saucelabs Grid

**Comment by diemol** [comment]
Are you having this issue? #9954

**Comment by lroman-uz** [comment]
Hola @diemol !
Is not that issue as we are not reaching our limit of VMs when I execute this test. I realized, and that's why I closed this issue, that this error appears depending on how much time it takes to download the files you put inside a prerun script for a Windows instance in Saucelabs. Just downloading a small file like a small picture it works 100% but if you are downloading bigger files likes videos or audios to mock the webcam it has a higher rate to fail, so I guess that this a issue related to Saucelabs side... a little bit weird as this is just happening with Selenium 4.
